go/libraries/doltcore/sqle/sqlbatch_test.go

```diff
@@ -188,29 +197,28 @@ func TestSqlBatchInsertIgnoreReplace(t *testing.T) {
 }
 
 func TestSqlBatchInsertErrors(t *testing.T) {
-	insertStatements := []string{
-		`insert into people (id, first, last, is_married, age, rating, uuid, num_episodes) values
-					(0, "Maggie", "Simpson", false, 1, 5.1, '00000000-0000-0000-0000-000000000007', 677)`,
-		`insert into people values
-					(2, "Milhouse", "VanHouten", false, 1, 5.1, true, 677)`,
-	}
-
 	dEnv := dtestutils.CreateTestEnv()
 	ctx := context.Background()
 
 	CreateTestDatabase(dEnv, t)
 	root, _ := dEnv.WorkingRoot(ctx)
 
-	batcher := NewSqlBatcher(dEnv.DoltDB, root)
-	for i := range insertStatements {
-		stmt := insertStatements[i]
-		statement, err := sqlparser.Parse(stmt)
-		require.NoError(t, err)
-		insertStmt, ok := statement.(*sqlparser.Insert)
-		require.True(t, ok)
-		_, err = ExecuteBatchInsert(context.Background(), root, insertStmt, batcher)
-		require.Error(t, err)
-	}
+	engine := sqle.NewDefault()
+	db := NewBatchedDatabase("dolt", root, dEnv)
+	engine.AddDatabase(db)
+
+	_, rowIter, err := engine.Query(sql.NewEmptyContext(), `insert into people (id, first, last, is_married, age, rating, uuid, num_episodes) values
+					(0, "Maggie", "Simpson", false, 1, 5.1, '00000000-0000-0000-0000-000000000007', 677)`)
+	// This won't generate an error until we commit the batch (duplicate key)
+	assert.NoError(t, err)
+	assert.NoError(t, drainIter(rowIter))
+
+	_, _, err = engine.Query(sql.NewEmptyContext(), `insert into people values
+					(2, "Milhouse", "VanHouten", false, 1, 5.1, true, 677)`)
+	assert.Error(t, err)
```

Comment:
The above does not error but this does before the flush? Probably something I'm overlooking...

Reply:
Added a clarifying comment